pkg/health/checker/checker.go

```diff
@@ -42,8 +46,7 @@ func NewConsulHealthChecker(opts kp.Options) ConsulHealthChecker {
 	client := kp.NewConsulClient(opts)
 	return consulHealthChecker{
 		client:      client,
-		health:      client.Health(),
-		consulStore: kp.NewConsulStore(opts),
+		consulStore: kp.NewConsulStore(client),
 		WaitTime:    1 * time.Minute,
```

Comment:
with these changes, is the `client` field still necessary? Also similar to what you did for kp.Store, i'd change this to consume the client directly (if it even still needs it). WaitTime argument should also be safe to remove.